main/observing/condition_checker.py

```diff
@@ -212,7 +212,9 @@ def rain_check(self):
                 urllib3.exceptions.InvalidHeader, requests.exceptions.ConnectionError, requests.exceptions.Timeout,
                 requests.exceptions.HTTPError):
             return None
-        api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+
+        #api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+        api_key = self.radar.text.split(r'"SUN_V3_API_KEY_CLIENT\":\"')[1].split(r'\",\"WX_API_PROTOCOL_CLIENT\"')[0]
```

Comment:
if you do a raw string (by doing r'foo' instead of just 'foo') then you shouldn't use the backslashes for the quotation marks...see the commented out string as a reference.

Reply:
Also I'm not sure what you're doing with the splits?  I would keep using re.search instead, unless you found that that didn't work for some reason.  Remember we still need some sort of fail safe in case it changes again, so you should save the .group() for after an if statement to make sure api_key exists.  If it doesn't then return or something.